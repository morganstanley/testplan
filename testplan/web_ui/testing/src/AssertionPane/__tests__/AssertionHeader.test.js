import React from "react";
import { render } from "@testing-library/react";
import { StyleSheetTestUtils } from "aphrodite";

import AssertionHeader from "../AssertionHeader";

const props = {
  assertion: {
    utc_time: "2024-12-24T09:35:56.113946+00:00",
    machine_time: "2024-12-24T17:35:56.113951+08:00",
    type: "Equal",
    meta_type: "assertion",
    description: "soda water == soda water",
    line_no: 26,
    category: "DEFAULT",
    flag: "DEFAULT",
    file_path: "/dummy/examples/Driver/Dependency/suites.py",
    passed: true,
    first: "soda water",
    second: "soda water",
    label: "==",
    type_actual: "str",
    type_expected: "str",
  },
  displayPath: false,
  uid: "c260c2af-888a-4853-86b2-82a14f44066e", // generated
  toggleExpand: jest.fn(),
  showStatusIcons: false,
};

describe("AssertionHeader", () => {
  let component;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  it("renders the correct HTML structure", () => {
    component = render(<AssertionHeader {...props} />);
    expect(component.getByText(props.assertion.description)).toBeDefined();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("renders the correct HTML structure with status icons enabled", () => {
    component = render(
      <AssertionHeader {...{ ...props, showStatusIcons: true }} />
    );
    expect(
      component.container.querySelector("svg").getAttribute("data-icon")
    ).toEqual("check");
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("renders the correct HTML structure with code context enabled", () => {
    component = render(
      <AssertionHeader {...{ ...props, displayPath: true }} />
    );
    expect(
      component.getByText(props.assertion.file_path, { exact: false })
    ).toBeDefined();
    expect(component.asFragment()).toMatchSnapshot();
  });
});
