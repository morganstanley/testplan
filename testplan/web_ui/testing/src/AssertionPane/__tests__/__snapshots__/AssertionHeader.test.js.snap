// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssertionHeader renders the correct HTML structure 1`] = `
<DocumentFragment>
  <div
    class="cardHeader_1s246z6-o_O-cardHeaderColorPassed_1nxwz58 card-header"
  >
    <div
      style="display: flex;"
    >
      <span
        class="button_13xlah4"
        style="order: 4; flex-grow: 4; padding: .125rem 0.75rem; display: flex; align-items: center;"
      >
        <span
          style="font-weight: bold;"
        >
          soda water == soda water 
        </span>
        <span>
          (Equal)
        </span>
      </span>
      <span
        class="cardHeaderAlignRight_107ja4p"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`AssertionHeader renders the correct HTML structure with code context enabled 1`] = `
<DocumentFragment>
  <div
    class="cardHeader_1s246z6-o_O-cardHeaderColorPassed_1nxwz58 card-header"
  >
    <div
      style="display: flex;"
    >
      <span
        class="button_13xlah4"
        style="order: 4; flex-grow: 4; padding: .125rem 0.75rem; display: flex; align-items: center;"
      >
        <span
          style="font-weight: bold;"
        >
          soda water == soda water 
        </span>
        <span>
          (Equal)
        </span>
      </span>
      <span
        class="cardHeaderAlignRight_107ja4p"
      />
      <div
        class="cardHeaderCode_658wrm-o_O-cardHeaderAlignRight_107ja4p-o_O-timeInfo_1kwowkq"
        style="order: 6; margin-left: 10px;"
      >
        <span>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text pathButton_vopdnm"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="cardHeaderColorPassed_1nxwz58"
                id="tooltip_path_c260c2af-888a-4853-86b2-82a14f44066e"
              >
                <span
                  class="icon_erbgzi"
                >
                  <i
                    class="fa fa-copy fa-s"
                  />
                </span>
                <div
                  class="cardHeaderPath_1b81ih6"
                >
                  <span
                    class="cardHeaderPathInner_1mceyef"
                  >
                    /dummy/examples/Driver/Dependency/suites.py:26
                  </span>
                </div>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <br />
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AssertionHeader renders the correct HTML structure with status icons enabled 1`] = `
<DocumentFragment>
  <div
    class="cardHeader_1s246z6-o_O-cardHeaderColorPassed_1nxwz58 card-header"
  >
    <div
      style="display: flex;"
    >
      <span
        class="button_13xlah4"
        style="order: 4; flex-grow: 4; padding: .125rem 0.75rem; display: flex; align-items: center;"
      >
        <span
          class="statusIcon_1xnyjf5"
        >
          <svg
            aria-labelledby="svg-inline--fa-title-1"
            class="svg-inline--fa fa-check fa-w-16 fa-sm icon_erbgzi"
            data-icon="check"
            data-prefix="fas"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="svg-inline--fa-title-1"
            >
              passed
            </title>
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          style="font-weight: bold;"
        >
          soda water == soda water 
        </span>
        <span>
          (Equal)
        </span>
      </span>
      <span
        class="cardHeaderAlignRight_107ja4p"
      />
    </div>
  </div>
</DocumentFragment>
`;
